tests/test.cpp

```diff
@@ -4324,6 +4324,108 @@ void FlatbuffersIteratorsTest() {
 void FlatbuffersIteratorsTest() {}
 #endif
 
+void PrivateAnnotationsLeaks() {
+  // Simple schemas and a "has optional scalar" sentinal.
+  std::vector<std::string> schemas;
+  std::vector<std::string> failure_schemas;
+
+  schemas.push_back(
+      "table Monster (private) { mana: int; }"
+      "struct ABC (private) { mana: int; }");
+  schemas.push_back(
+      "table Monster { mana: int; }"
+      "struct ABC { mana: int; }");
+
```

Comment:
nit: The spacing between schemas is inconsistent.

Optional: it might also be a good idea to add comments naming/describing the different cases

I'd suggest a comment scheme matching

(private)? (table|struct|union) containing (private)? (struct|table|enum|union)

Reply:
Would this be good enough:

```
public (all)
schemas.push_back(
    "table Monster { mana: int; }"
    "struct ABC { mana: int; }");

public (ABC) private (Monster)
schemas.push_back(
    "table Monster (private) { mana: int; }"
    "struct ABC { mana: int; }");
```

Reply:
I'm not sure what you mean with that

My suggestion was to add comments that describe cases are being tested. I think the cases are based on the _kinds_ of types that flatbuffers supports (enum/struct/union/etc). The _names_ of the types (ABC/Monster) aren't relevant to the cases that we're trying to test